patch.go

```diff
@@ -359,7 +359,7 @@ func findObject(pd *container, path string) (container, string) {
 
 		next, ok := doc.get(decodePatchKey(part))
 
-		if next == nil || ok != nil {
```

Comment:
My guess is that it would be better to correct this variable name from `ok` to `err`, and fix https://github.com/evanphx/json-patch/blob/master/patch.go#L395 to return an error if the key doesn't exist.

A test case demonstrating how to produce this condition would be good?